software/winrm/src/test/java/org/apache/brooklyn/util/core/internal/winrm/winrm4j/PrettyXmlWriterTest.java

```diff
@@ -134,6 +134,24 @@ public void testIgnoreComment() throws IOException {
         assertEquals(recordingWriter.out.toString(), "#< CLIXML\n<t1>\n\t<t2>\n\t\t<t3/>\n\t</t2>\n</t1>");
     }
 
+    @Test
+    public void testWriteInChunks() throws IOException {
+        String xml = "<t1><t2>A</t2></t1>";
+        for (int breakAt = 0; breakAt < xml.length(); breakAt++) {
+            if (breakAt == 15) continue;
```

Comment:
There is one more test-case, forgot to verify it. Code change for this to follow.

Reply:
This is addressed in the new commit.